test/core/vector/Vector_test.cpp

```diff
@@ -300,15 +294,17 @@ void runTests(int argc, char **argv)
         rob_tester.test_num_insts_retired(2);
         rob_tester.test_num_uops_retired(5);
         rob_tester.test_last_inst_has_tail(false);
+
+        EXPECT_TRUE(my_vuop_generator->vuops_generated_.get() == 4);
```

Comment:
Can you add this check to all of the vector tests? You can do it in a separate PR after you release this one.